test/Microsoft.SqlTools.ServiceLayer.Test/LanguageServer/PeekDefinitionTests.cs

```diff
@@ -0,0 +1,236 @@
+        {
+            // Get live connectionInfo
+            ConnectionInfo connInfo = TestObjects.InitLiveConnectionInfoForDefinition();
+            PeekDefinition peekDefinition = new PeekDefinition(connInfo);
+            string objectName = "test_table";
+            string schemaName = null;
+            string objectType = "TABLE";
+
+            Location[] locations = peekDefinition.GetSqlObjectDefinition(peekDefinition.GetTableScripts, objectName, schemaName, objectType);
+            Assert.NotNull(locations);
+            Cleanup(locations);
+        }
+
+        /// <summary>
+        /// Test get definition for a view object with active connection
+        /// </summary>
+        [Fact]
+        public void GetViewDefinitionTest()
+        {
+            ConnectionInfo connInfo = TestObjects.InitLiveConnectionInfoForDefinition();
+            PeekDefinition peekDefinition = new PeekDefinition(connInfo);           
+            string objectName = "objects";
+            string schemaName = "sys";
+            string objectType = "VIEW";
+
+            Location[] locations = peekDefinition.GetSqlObjectDefinition(peekDefinition.GetViewScripts, objectName, schemaName, objectType);
+            Assert.NotNull(locations);
+            Cleanup(locations);
+        }
+
+        /// <summary>
+        /// Test get definition for an invalid view object with active connection
+        /// </summary>
+        [Fact]
+        public void GetViewDefinitionInvalidObjectTest()
+        {
+            ConnectionInfo connInfo = TestObjects.InitLiveConnectionInfoForDefinition();
+            PeekDefinition peekDefinition = new PeekDefinition(connInfo);
+            string objectName = "objects";
+            string schemaName = null;
+            string objectType = "VIEW";
+
+            Location[] locations = peekDefinition.GetSqlObjectDefinition(peekDefinition.GetViewScripts, objectName, schemaName, objectType);
+            Assert.Null(locations);
+        }
+
+        /// <summary>
+        /// Test get definition for a stored procedure object with active connection
+        /// </summary>
+        [Fact]
+        public void GetStoredProcedureDefinitionTest()
+        {
+            ConnectionInfo connInfo = TestObjects.InitLiveConnectionInfoForDefinition();
+            PeekDefinition peekDefinition = new PeekDefinition(connInfo);
+            string objectName = "SP1";
+            string schemaName = "dbo";
+            string objectType = "PROCEDURE";
+
+            Location[] locations = peekDefinition.GetSqlObjectDefinition(peekDefinition.GetStoredProcedureScripts, objectName, schemaName, objectType);
+            Assert.NotNull(locations);
+            Cleanup(locations);
+        }
+
+        /// <summary>
+        /// Test get definition for a stored procedure object with active connection and no schema
+        /// </summary>
+        [Fact]
+        public void GetStoredProcedureDefinitionWithoutSchemaTest()
```

Comment:
Where is the test covering how we handle an unsupported object type? It is important to test negative scenarios and verify we don't crash but instead return empty results cleanly. WIth the new GetSqlObjectDefinition this should be possible? Or is this covered in a different set of tests?